File: api-backend/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace api.Models
{
    [Table("Posts")]
    public class Post
    {
        [Key]
        public Guid Id { get; set; }
        public string AppUserId { get; set; }
        public string Tema { get; set; } = string.Empty;
        public string PostContent { get; set; } = string.Empty;
        public DateTime CreatedOn { get; set; }
        public List<Comentario> Comentarios { get; set; } = [];
        public List<Avaliacao> Avaliacoes { get; set; } = [];
        public AppUser AppUser { get; set; }
    }
}

File: frontend/src/Models/Post.ts
import { avaliacao } from "./Avaliacao";
import { Comentario } from "./Comment";
import { User } from "./User";

export interface Post {
  id: string;
  tema: string;
  postContent: string;
  createdOn: string;
  avaliacoes: avaliacao[];
  comentarios: Comentario[];
  appUserId: string;
  appUser: User;
}

export interface CreatePostDto {
  title: string;
  content: string;
}

export interface UpdatePostRequestDto {
  title: string;
  content: string;
}

export interface PagedResponse<T> {
  data: T;
  PageNumber: number;
  PageSize: number;
  TotalPages: number;
  TotalRecords: number;
  isFirstpage: boolean;
  isLastpage: boolean;
  
}

export interface GenericPaginationModel<T> {
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
  Value: T;
}

export interface PainelForum {
  search: string;
  avaliacao: number;
  IsDescending: boolean;
}
